### PR TITLE
Protect against server hijacking error handling

### PR DESCRIPTION
**Pull request by paulocsanz**
If a server returns "HTTP/1.x -8 OK", for example, it can misguide an application developer into freeing less-important memory so the request can be retried and succeed, when the problem is in the server.

_returnCode is never used anywhere else, but it could still contain a negative value returned by a broken server and therefore could cause troubles in the future (if _returnCode is in fact used)

### CODE REVIEW

File: libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp [hunk 1]
```diff
@@ -1125,17 +1125,17 @@ int HTTPClient::handleHeaderResponse()
                     if(transferEncoding.equalsIgnoreCase(F("chunked"))) {
                         _transferEncoding = HTTPC_TE_CHUNKED;
                     } else {
-                        return HTTPC_ERROR_ENCODING;
+                        return (_returnCode = HTTPC_ERROR_ENCODING);
```
Comment (earlephilhower) [inline libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp:1128]: Style request:  Please separate assignment and return here and below, just for sanity's sake.

**Review by earlephilhower — CHANGES_REQUESTED** [review]
Have you actually seen HTTP servers so far out of spec that they really return negative error codes? 

Minor style change, please.  OTW seems fine.

File: libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp (hunk 1 above)
Comment (paulocsanz) [reply libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp:1128]: Done!

**Comment by paulocsanz** [comment]
No, I've not. It's probably not a big deal. It was just something I caught in a code "audit".

**Review by earlephilhower — APPROVED** [review]
Thanks!  LGTM.

**Pull request merged**